roles/teslamate/defaults/main.yml

```diff
@@ -0,0 +1,158 @@
+# Title:            Sandbox: teslamate                                  #
+# Author(s):        maindust                                            #
+# URL:              https://github.com/saltyorg/Sandbox                 #
+# --                                                                    #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+# defaults/main.yml
+
+teslamate_name: teslamate
+
+################################
+# Settings
+################################
+
+TM_DB_USER: "{{ user.name }}"
+TM_DB_PASS: "password4321"
+TM_DB_NAME: "teslamate"
+TESLAMATE_SECRET_KEY: "{{ teslamate_saltbox_facts.facts.secret_key }}"
+
+################################
+# Paths
+################################
+
+teslamate_paths_folder: "{{ teslamate_name }}"
+teslamate_paths_location: "{{ server_appdata_path }}/{{ teslamate_paths_folder }}"
+teslamate_paths_folders_list:
+  - "{{ teslamate_paths_location }}"
+
+################################
+# Web
+################################
+
+teslamate_web_subdomain: "{{ teslamate_name }}"
+teslamate_web_domain: "{{ user.domain }}"
+teslamate_web_port: "4000"
+teslamate_web_url: "{{ 'https://' + (teslamate_web_subdomain + '.' + teslamate_web_domain
+                  if (teslamate_web_subdomain | length > 0)
+                  else teslamate_web_domain) }}"
+
+################################
+# DNS
+################################
+
+teslamate_dns_record: "{{ teslamate_web_subdomain }}"
+teslamate_dns_zone: "{{ teslamate_web_domain }}"
+teslamate_dns_proxy: "{{ dns.proxied }}"
+
+################################
+# Traefik
+################################
+
+teslamate_traefik_sso_middleware: ""
+teslamate_traefik_middleware_default: "{{ traefik_default_middleware }}"
+teslamate_traefik_middleware_custom: ""
+teslamate_traefik_certresolver: "{{ traefik_default_certresolver }}"
+teslamate_traefik_enabled: true
+teslamate_traefik_api_enabled: false
+teslamate_traefik_api_endpoint: ""
+
+################################
+# Docker
+################################
+
+# Container
+teslamate_docker_container: "{{ teslamate_name }}"
+
+# Image
+teslamate_docker_image_pull: true
+teslamate_docker_image_tag: "latest"
+teslamate_docker_image: "teslamate/teslamate:{{ teslamate_docker_image_tag }}"
+
+# Ports
+teslamate_docker_ports_defaults: []
+teslamate_docker_ports_custom: []
+teslamate_docker_ports: "{{ teslamate_docker_ports_defaults
+                          + teslamate_docker_ports_custom }}"
+
+# Envs
+teslamate_docker_envs_default:
+  DATABASE_USER: "{{  TM_DB_USER }}"
+  DATABASE_PASS: "{{ TM_DB_PASS }}"
+  DATABASE_NAME: "{{ TM_DB_NAME }}"
+  ENCRYPTION_KEY: "{{ TESLAMATE_SECRET_KEY }}"
+  DATABASE_HOST: "postgres"         # Set the database host to the PostgreSQL container
```

Comment:
We should be provisioning a separate postgres instance just for Teslamate rather than using the generic one

Reply:
What’s the purpose of using a new Postgres instance vs the generic one? Not an issue, just curious

Reply:
Allows for granular version control between different apps, easier to reset an app config if needed